onecgiar-pr-server/src/api/notification/notification.service.ts

```diff
@@ -0,0 +1,449 @@
+      });
+
+      const whereConditions: WhereConditions = {
+        target_user: user.id,
+        read: false,
+        obj_result: {
+          is_active: true,
+          obj_result_by_initiatives: { initiative_role_id: 1 },
+        },
+      };
+
+      if (userLastViewed.last_pop_up_viewed) {
+        whereConditions.created_date = MoreThan(
+          userLastViewed.last_pop_up_viewed,
+        );
+      }
+
+      const notificationsUpdates = await this._notificationRepository.find({
+        select: this.getNotificattionSelect(),
+        relations: this.getNotificationRelations(),
+        where: whereConditions,
+      });
+
+      const shareResultPendings =
+        await this._shareResultRequestService.getReceivedResultRequestPopUp(
+          user,
+        );
+
+      const isError = (shareResultPendings as any)?.response;
+      const notifications = isError
+        ? notificationsUpdates
+        : [
+            ...notificationsUpdates,
+            ...(Array.isArray(shareResultPendings) ? shareResultPendings : []),
+          ];
+
+      return {
+        response: notifications,
+        message: 'List of all notifications retrieved successfully',
+        status: HttpStatus.OK,
+      };
+    } catch (error) {
+      this._logger.error(error);
+      return {
+        response: [],
+        message: 'An error occurred while retrieving the notifications',
+        status: HttpStatus.INTERNAL_SERVER_ERROR,
+      };
+    }
+  }
+
+  private getNotificattionSelect() {
+    return {
+      obj_notification_level: {
+        notifications_level_id: true,
+        type: true,
+      },
+      obj_notification_type: {
+        notifications_type_id: true,
+        type: true,
+      },
+      obj_emitter_user: {
+        id: true,
+        first_name: true,
+        last_name: true,
+        email: true,
+      },
+      obj_target_user: {
+        id: true,
+        first_name: true,
+        last_name: true,
+        email: true,
+      },
+      obj_result: {
+        result_code: true,
+        title: true,
+        status_id: true,
+        obj_result_by_initiatives: {
+          initiative_id: true,
+          obj_initiative: {
+            id: true,
+            official_code: true,
+          },
+        },
+        obj_version: {
+          id: true,
+          phase_name: true,
+        },
+      },
+    };
+  }
+
+  private getNotificationRelations() {
+    return {
+      obj_notification_level: true,
+      obj_notification_type: true,
+      obj_emitter_user: true,
+      obj_target_user: true,
+      obj_result: {
+        obj_result_by_initiatives: {
+          obj_initiative: true,
+        },
+        obj_version: true,
+      },
+    };
+  }
+}
+
+interface WhereConditions {
+  target_user: number;
+  read: boolean;
+  obj_result: {
+    is_active: boolean;
+    obj_result_by_initiatives: {
+      initiative_role_id: number;
+    };
+  };
+  created_date?: FindOperator<Date>;
+}
```

Comment:
@JuanGuzman-io consider using `FindManyOptions|FindOneOptions<your-entity>` instead

Reply:
@gcmartinezg implement the FindOptionsWhere instead of FindManyOptions|FindOneOptions<your-entity>